estate/models/estate_property.py

```diff
@@ -0,0 +1,167 @@
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        required=True,
+        copy=False,
+        default='new',
+    )
+    property_type_id = fields.Many2one(
+        "estate.property.type",
+        string="Property Type"
+    )
+
+    buyer_id = fields.Many2one(
+        "res.partner",
+        string="Buyer",
+        copy=False
+    )
+
+    salesperson_id = fields.Many2one(
+        "res.users",
+        string="Salesman",
+        default=lambda self: self.env.user
+    )
+
+    tag_ids = fields.Many2many(
+        "estate.property.tag",
+        string="Tags"
+    )
+
+    offer_ids = fields.One2many(
+        "estate.property.offer",
+        "property_id",
+        string="Offers",
+    )
+
+    total_area = fields.Float(
+        compute="_compute_total_area",
+        string="Total Area",
+    )
+
+    best_price = fields.Float(
+        compute="_compute_best_price",
+        string="Best Offer",
+    )
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_price = max(record.offer_ids.mapped("price"))
+            else:
+                record.best_price = 0.0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = "north"
+        else:
+            self.garden_area = 0
+            self.garden_orientation = False
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = (
+                    record.create_date.date() + timedelta(days=record.validity)
+            )
+            else:
+                record.date_deadline = fields.Date.context_today(self)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            if record.create_date and record.date_deadline:
+                record.validity = (
+                    record.date_deadline - record.create_date.date()
+                ).days
+
+    def action_cancel(self):
+        for record in self:
+            if record.state == "sold":
+                raise UserError("A sold property cannot be cancelled.")
+            record.state = "cancelled"
+        return True
+
+    def action_sold(self):
+        for record in self:
+            if record.state == "cancelled":
+                raise UserError("A cancelled property cannot be sold.")
+            record.state = "sold"
+        return True
+
+    @api.constrains("selling_price", "expected_price")
+    def _check_selling_price(self):
```

Comment:
```suggestion
    def _check_selling_price_value(self):
```
you already used the name `_check_selling_price` for a constraint earlier. While this technically *should* work, it’s confusing for maintenance.